### PR TITLE
_cluster/state should always return cluster_uuid

### PR DESCRIPTION
**Pull request by pickypg**
Currently, the only way to get the REST response for the `/_cluster/state`
call to return the `cluster_uuid` is to request the `metadata` metrics,
which is one of the most expensive response structures. However, external
monitoring agents will likely want the `cluster_uuid` to correlate the
response with other API responses whether or not they want cluster
metadata.

Relates to https://github.com/elastic/beats/pull/6807

### CODE REVIEW

**Review by bleskes — CHANGES_REQUESTED** [review]
LGTM, though I believe you need a skip clause in your yml tests to make sure it doesn't run against an old version that doesn't output the uuid.

**Comment by pickypg** [comment]
@bleskes I wonder if I should just make a specific test for the cluster_uuid in that case, then skip only that one for older versions and avoid having none of these REST tests run on older versions?

**Comment by bleskes** [comment]
@pickypg makes sense.

File: rest-api-spec/src/main/resources/rest-api-spec/test/cluster.state/10_basic.yml [hunk 1]
```diff
@@ -18,3 +18,18 @@
   - is_true: master_node
   - gte: { compressed_size_in_bytes: 50 }
   - is_true: compressed_size
+
+---
+"get cluster state returns cluster_uuid at the top level":
+  - skip:
+      version:  " - 6.99.99"
```
Comment (pickypg) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cluster.state/10_basic.yml]: There will be a follow-up PR to change this to `6.3.99` after this PR is merged / backported to 6.x (6.4).

**Comment by pickypg** [comment]
jenkins test this

**Comment by pickypg** [comment]
jenkins test this

**Comment by pickypg** [comment]
@bleskes Updated with skip logic and CI has calmed down

**Review by bleskes — APPROVED** [review]

File: rest-api-spec/src/main/resources/rest-api-spec/test/cluster.state/10_basic.yml (hunk 1 above)
Comment (pickypg) [reply rest-api-spec/src/main/resources/rest-api-spec/test/cluster.state/10_basic.yml]: https://github.com/elastic/elasticsearch/pull/30264 as promised

**Pull request merged**